lib/iris/tests/unit/analysis/scipy_interpolate/test__RegularGridInterpolator.py

```diff
@@ -0,0 +1,85 @@
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for the :func:`iris.analysis.interpolate.linear` function."""
+
+from __future__ import (absolute_import, division, print_function)
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+import mock
+
+import numpy as np
+import iris
+
+from iris.analysis._scipy_interpolate import _RegularGridInterpolator
+from scipy.sparse.csr import csr_matrix
+import iris.tests.stock as stock
+
+
+class Test(tests.IrisTest):
+    def setUp(self):
+        '''Load a source cube, then generate an interpolator instance and
+        target grid.'''
+        self.cube = stock.simple_2d()
+        x_points = self.cube.coord('bar').points
+        y_points = self.cube.coord('foo').points
+        self.interpolator = _RegularGridInterpolator([x_points, y_points],
+                                                     self.cube.data,
+                                                     method='linear',
+                                                     bounds_error=False,
+                                                     fill_value=None)
+        newx = x_points + 0.7
+        newy = y_points + 0.7
+        self.test_increment = 0.60666667
+        tgrid = np.empty((len(newx), len(newy), 2))
+        values = []
+        for xs in newx:
+            for ys in newy:
+                values.extend([xs, ys])
+        for i, point in enumerate(np.nditer(tgrid, op_flags=['readwrite'])):
+            point[...] = values[i]
+        self.interp_coords = tgrid
+
+    def test_compute_interp_weights(self):
+        interpolator = self.interpolator
+        weights = interpolator.compute_interp_weights(self.interp_coords)
+        self.assertIsInstance(weights, tuple)
+        self.assertEqual(len(weights), 5)
+        self.assertEqual(weights[0], self.interp_coords.shape)
+        self.assertEqual(weights[1], 'linear')
+        self.assertIsInstance(weights[2], csr_matrix)
+
+    def test__evaluate_linear_sparse(self):
+        interpolator = self.interpolator
+        weights = interpolator.compute_interp_weights(self.interp_coords)
+        output_data = interpolator._evaluate_linear_sparse(weights[2])
+        test_data = self.cube.data.reshape(-1) + self.test_increment
+        self.assertArrayAlmostEqual(output_data, test_data)
+
+    def test_interp_using_pre_computed_weights(self):
+        interpolator = self.interpolator
+        weights = interpolator.compute_interp_weights(self.interp_coords)
```

Comment:
These two lines are common to all three of your tests. Suggest shifting the line that calculates the weights into `setUp` and setting weights as a attribute of the class, and dropping the first line entirely.